workers/unity/Packages/com.improbable.gdk.core/Utility/PlayerLoopUtils.cs

```diff
@@ -0,0 +1,238 @@
+            var subSystemToGroup = new Dictionary<Type, List<ComponentSystemGroup>>();
+
+            // Build lookup for PlayerLoop
+            foreach (var systemGroup in systemGroups)
+            {
+                var type = systemGroup.GetType();
+                Type subSystemType;
+
+                // Hardcode the subSystem types for Unity root-systems
+                if (type == typeof(InitializationSystemGroup))
+                {
+                    subSystemType = typeof(Initialization);
+                }
+                else if (type == typeof(SimulationSystemGroup))
+                {
+                    subSystemType = typeof(Update);
+                }
+                else if (type == typeof(PresentationSystemGroup))
+                {
+                    subSystemType = typeof(PreLateUpdate);
+                }
+                else
+                {
+                    var attributes = type.GetCustomAttributes(typeof(UpdateInSubSystemAttribute), true);
+                    if (attributes.Length == 0)
+                    {
+                        continue;
+                    }
+
+                    subSystemType = ((UpdateInSubSystemAttribute) attributes[0]).SubSystemType;
+                }
+
+                // Add to lookup table
+                if (!subSystemToGroup.TryGetValue(subSystemType, out var groupList))
+                {
+                    groupList = new List<ComponentSystemGroup>();
+                    subSystemToGroup.Add(subSystemType, groupList);
+                }
+
+                groupList.Add(systemGroup);
+            }
+
+            // Insert groups into PlayerLoop
+            var playerLoop = ScriptBehaviourUpdateOrder.CurrentPlayerLoop;
+            if (playerLoop.subSystemList == null)
+            {
+                playerLoop = PlayerLoop.GetDefaultPlayerLoop();
+            }
+
+            for (var i = 0; i < playerLoop.subSystemList.Length; ++i)
+            {
+                ref var subSystem = ref playerLoop.subSystemList[i];
+                if (!subSystemToGroup.TryGetValue(subSystem.type, out var groupList))
+                {
+                    // No groups to add for this subsystem
+                    continue;
+                }
+
+                var originalSize = subSystem.subSystemList.Length;
+                var newSize = originalSize + groupList.Count;
+                Array.Resize(ref subSystem.subSystemList, newSize);
+                for (var groupIndex = 0; groupIndex < groupList.Count; groupIndex++)
+                {
+                    var systemGroup = groupList[groupIndex];
+                    InsertManagerIntoSubsystemList(subSystem.subSystemList, originalSize + groupIndex,
+                        systemGroup, systemGroup.GetType());
+                }
+            }
+
+            // Set as new PlayerLoop
+            ScriptBehaviourUpdateOrder.SetPlayerLoop(playerLoop);
+        }
+
+        public static void PrintPlayerLoop()
```

Comment:
Was this for debugging? Or still needed?

Reply:
It's for debugging, I'd like to keep it around, as there is no way to print the whole player loop.